compiler/AST/symbol.cpp

```diff
@@ -303,6 +308,82 @@ bool VarSymbol::isParameter() const {
   return hasFlag(FLAG_PARAM) || immediate;
 }
 
+
+std::string VarSymbol::docsDirective() {
+  std::string result;
+  if (fDocsTextOnly) {
+    result = "";
+  } else {
+    // TODO: If this is a type (i.e. this->hasFlag(FLAG_TYPE_VARIABLE)), use
+    //       ".. type:: " as directive. (thomasvandoren, 2015-02-21)
+    if (this->isField) {
+      result = ".. attribute:: ";
+    } else {
+      result = ".. data:: ";
+    }
+  }
+  return this->hasFlag(FLAG_CONFIG) ? result + "config " : result;
+}
+
+
+void VarSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC) || this->hasFlag(FLAG_SUPER_CLASS)) {
+      return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  if (this->hasFlag(FLAG_TYPE_VARIABLE)) {
+    *file << "type ";
+  } else if (this->isConstant()) {
+    *file << "const ";
+  } else if (this->isParameter()) {
+    *file << "param ";
+  } else {
+    *file << "var ";
+  }
+
+  *file << this->name;
+
+  if (this->defPoint->exprType != NULL) {
+    *file << ": ";
+    this->defPoint->exprType->prettyPrint(file);
+  }
+
+  // FIXME: Do we want this for non-class globals/configs?
+  //        (thomasvandoren, 2015-02-21)
```

Comment:
In both languages, it's useful to have type aliases for the benefit of creating symbolic names and have the flexibility of mapping the type to different representations over time or across implementations.  In both languages, it's useful to know what the type actually maps down to so that you can do things with it that are legal and avoid things that are illegal (for example, "What control string should I use when I printf() this size_t?"  "Can I add 1 to this size_t?"  "If I overload a function on this type, will it conflict with other overloads?").

I posit that if C had well-defined int sizes, their interfaces would be defined far more often in terms of base types rather than special types like size_t -- that is, I think most of these type aliases cases are designed to deal with ambiguities in the integer sizes.  I expect Chapel to require a lot fewer typedefs in these types of situations (and more generally, fewer module-scoped typedefs) for the same reason.

But I suggest we continue supporting module-level type aliases, and documenting them, and documenting what they map to in spite of this prediction.